app/Svg/Parser.hs

```diff
@@ -286,6 +289,39 @@ readAttr attr tag = fromMaybe
     (error ("reading " ++ T.unpack attr ++ " from " ++ show tag))
     (readMaybe $ T.unpack $ fromAttrib attr tag)
 
+-- | Runs a parser on a text object.
+-- Throws an exception on any parse errors.
+parseVal :: Parser a -> T.Text -> a
+parseVal parser input =
+    case P.parse parser "" strippedInput of
+        Left err ->
+            error ("reading " ++ T.unpack input ++ ":" ++ show err)
+        Right val -> val
+    where strippedInput = filter (not . isSpace) (T.unpack input)
+
+-- | Looks up an attribute value using the given parser.
+parseAttr :: Parser a -> T.Text     -- ^ The attribute's name.
+                      -> Tag T.Text -- ^ The element containing the attribute.
+                      -> a
+parseAttr parser attr tag = parseVal parser $ fromAttrib attr tag
+
+-- | Parses one or more digit characters.
+digits :: Parser String
+digits = P.many1 P.digit
+
+-- | Parses a double value, ignoring units if present.
+double :: Parser Double
+double = do
+    sign <- P.option "" (P.string "-")
```

Comment:
Use `char` from `Text.Parsec.Char` for single characters (same with the comma below).